parser/quakec-parser.js

```diff
@@ -1343,10 +1343,26 @@ var parse = function(programInfo) {
 
     lexer.setInput(programInfo.program);
     new Scope(programInfo.uri, programInfo.parentScope);
-    Parse.advance();
-    let ast = Parse.definitions();
-    let scope = Context.scope;
-    Context.scope.pop()
+
+    let ast, scope;
+
+    try
+    {
+        Parse.advance();
+        ast = Parse.definitions();
+        scope = Context.scope;
+        Context.scope.pop();
+    }
+    catch (e)
+    {
+        Context.errors.push(
```

Comment:
This is okay for now. But ideally the parser should be robust enough to handle whatever is thrown at it and not crash.